connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -3051,202 +2238,174 @@ public void testKeyRotationDisabledWhenWorkerBecomesFollower() throws Exception
                 Collections.emptySet(),
                 Collections.emptySet());
         expectConfigRefreshAndSnapshot(snapshotWithKey);
-        // First rebalance: poll for a limited time as worker is leader and must wake up for key expiration
-        member.poll(leq(rotationTtlDelay));
-        EasyMock.expectLastCall();
-
-        expectRebalance(1, Collections.emptyList(), Collections.emptyList());
-        // Second rebalance: poll indefinitely as worker is no longer leader, so key expiration doesn't come into play
-        member.poll(Long.MAX_VALUE);
-        EasyMock.expectLastCall();
-
-        PowerMock.replayAll(initialSecretKey);
+        doNothing().when(member).poll(anyLong());
 
         configUpdateListener.onSessionKeyUpdate(initialKey);
         herder.tick();
+
+        expectRebalance(1, Collections.emptyList(), Collections.emptyList());
         herder.tick();
 
-        PowerMock.verifyAll();
+        // First rebalance: poll for a limited time as worker is leader and must wake up for key expiration
+        verify(member).poll(leq(rotationTtlDelay));
+        // Second rebalance: poll indefinitely as worker is no longer leader, so key expiration doesn't come into play
+        verify(member).poll(eq(Long.MAX_VALUE));
+        verifyNoMoreInteractions(worker, member, configBackingStore, statusBackingStore);
     }
 
     @Test
     public void testPutTaskConfigsSignatureNotRequiredV0() {
-        Callback<Void> taskConfigCb = EasyMock.mock(Callback.class);
-
-        member.wakeup();
-        EasyMock.expectLastCall().once();
-        EasyMock.expect(member.currentProtocolVersion()).andReturn(CONNECT_PROTOCOL_V0).anyTimes();
-        PowerMock.replayAll(taskConfigCb);
+        when(member.currentProtocolVersion()).thenReturn(CONNECT_PROTOCOL_V0);
 
+        Callback<Void> taskConfigCb = mock(Callback.class);
         herder.putTaskConfigs(CONN1, TASK_CONFIGS, taskConfigCb, null);
 
-        PowerMock.verifyAll();
+        // Expect a wakeup call after the request to write task configs is added to the herder's request queue
+        verify(member).wakeup();
```

Comment:
We also have to make sure that the callback wasn't completed with an error:

```suggestion
        verify(member).wakeup();
        verifyNoMoreInteractions(taskConfigCb);
```

(This also applies to `testPutTaskConfigsSignatureNotRequiredV1`, and would probably be useful in all the other request signing test cases)